src/include/gcc.h

```diff
@@ -131,18 +131,37 @@ __wt_atomic_cas_ptr(void *vp, void *old, void *newv)
-    static inline ret __wt_atomic_fetch_add##name(vp_arg, v_arg) \
-    {                                                            \
-        return (__atomic_fetch_add(vp, v, __ATOMIC_SEQ_CST));    \
-    }                                                            \
-    static inline ret __wt_atomic_sub##name(vp_arg, v_arg)       \
-    {                                                            \
-        return (__atomic_sub_fetch(vp, v, __ATOMIC_SEQ_CST));    \
+#define WT_ATOMIC_FUNC(name, ret, vp_arg, v_arg)                                                 \
+    static inline ret __wt_atomic_add##name(vp_arg, v_arg)                                       \
+    {                                                                                            \
+        return (__atomic_add_fetch(vp, v, __ATOMIC_SEQ_CST));                                    \
+    }                                                                                            \
+    static inline ret __wt_atomic_fetch_add##name(vp_arg, v_arg)                                 \
+    {                                                                                            \
+        return (__atomic_fetch_add(vp, v, __ATOMIC_SEQ_CST));                                    \
+    }                                                                                            \
+    static inline ret __wt_atomic_sub##name(vp_arg, v_arg)                                       \
+    {                                                                                            \
+        return (__atomic_sub_fetch(vp, v, __ATOMIC_SEQ_CST));                                    \
+    }                                                                                            \
+    /*                                                                                           \
+     * !!! Attention !!!                                                                         \
+     * The following atomic functions are ATOMIC_RELAXED, while the preceding calls are          \
+     * ATOMIC_SEQ_CST. Mixing RELAXED and SEQ_CST means we will *not* get SEQ_CST guarantees     \
+     * when mixing these functions.                                                              \
+     * Historically WiredTiger mixed the SEQ_CST calls above with non-atomic accesses to memory, \
+     * and these non-atomic calls are being replaced with atomic calls. Using these new atomic   \
+     * functions with SEQ_CST memory ordering comes with a moderate performance cost, so we're   \
+     * using RELAXED to avoid the performance loss. In future these atomics should be updated to \
+     * all use the same memory ordering.                                                         \
+     */                                                                                          \
+    static inline ret __wt_atomic_load##name(vp_arg)                                             \
```

Comment:
These macros are using `ATOMIC_RELAXED` as [discussed in WT-12343](https://jira.mongodb.org/browse/WT-12343?focusedCommentId=6077625&focusedId=6077625&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-6077625)